Habitica/src/main/java/com/habitrpg/android/habitica/helpers/FirstDayOfTheWeekHelper.java

```diff
@@ -8,34 +8,39 @@
 public class FirstDayOfTheWeekHelper {
 
     private int firstDayOfTheWeek;
+    private int dailyTaskFormOffset;
 
-    public FirstDayOfTheWeekHelper(String firstDayOfTheWeekSharedPref) {
+    private FirstDayOfTheWeekHelper(int dailyTaskFormOffset, int firstDayOfTheWeek) {
+        this.dailyTaskFormOffset = dailyTaskFormOffset;
+        this.firstDayOfTheWeek = firstDayOfTheWeek;
+    }
+
+    public static FirstDayOfTheWeekHelper newInstance(String firstDayOfTheWeekSharedPref) {
```

Comment:
In this specific class, it decouples the logic of the class from the constructor which allows for easier testing. It is worth noting that I could have also achieved this by adding a new method and placed the logic there. This, however, would require two calls, object instantiation & method call, every time the class is used in the app. In general, static factory methods for object creation are worth considering because: One, unlike constructors, they have names, so instead of newInstance you could use very specific names describing the purpose of the constructor. Something that sometimes the method signature cannot achieve on its own. Two, unlike constructors, they are not required to create a new object each time they’re invoked, meaning you could create a static factory that returns the same instance every time whereas when you use new ClassName() you'll always return a new instance. There is a more in-depth explanation of static factory methods in Effective Java, 2nd edition.